src/types/params.ts

```diff
@@ -93,7 +90,7 @@ export type ExtractRouteParamTypes<TRoute> = TRoute extends {
  * @returns A new type with the appropriate properties marked as optional.
  */
 export type ExtractParamTypes<TParams extends Record<string, Param>> = Identity<MakeOptional<{
-  [K in keyof TParams]: ExtractParamType<TParams[K]>
+  [K in keyof TParams as ExtractParamName<K>]: ExtractParamType<TParams[K], K>
```

Comment:
this does 2 important things.
- it changes the key to be the name of the param without `?` for devs
- it passes in the unmodified `K` to be used by `ExtractParamType` to determine if type should be optional